hazelcast/src/main/java/com/hazelcast/jet/core/processor/SourceProcessors.java

```diff
@@ -443,7 +444,8 @@ public static <T> ProcessorMetaSupplier readJdbcP(
             @Nonnull ToResultSetFunction resultSetFn,
             @Nonnull FunctionEx<? super ResultSet, ? extends T> mapOutputFn
     ) {
-        return ReadJdbcP.supplier(context -> new DataSourceFromConnectionSupplier(newConnectionFn), resultSetFn, mapOutputFn);
+        return ReadJdbcP.supplier(context -> DataStoreHolder.closing(new DataSourceFromConnectionSupplier(newConnectionFn)),
```

Comment:
One more place, wrapping seems unneeded, since `DataSourceFromConnectionSupplier` is not Closable.

Reply:
fixed https://github.com/hazelcast/hazelcast/pull/22372/commits/d7fc71874c0595930eba531be3c8ddc05b4623bd